slackboard/notify.go

```diff
@@ -78,21 +82,57 @@ func sendNotification2Slack(payload *SlackPayload, sync bool) (int, error) {
 		Timeout: 30 * time.Second,
 	}
 
-	resp, err := client.Post(
-		ConfSlackboard.Core.SlackURL,
-		"application/json",
-		strings.NewReader(string(body)))
+	if len(ConfSlackboard.Core.SlackToken) == 0 {
+		// if SlackToken is not specified, use Incoming Webhook
+		resp, err := client.Post(
+			ConfSlackboard.Core.SlackURL,
+			"application/json",
+			bytes.NewReader(body),
+		)
+		if err != nil {
+			return http.StatusBadGateway, err
```

Comment:
Why the status code is fixed with 502?

Reply:
Here inside `if len(SlackToken) == 0` block implements backward compatibility.
This line should return 502, even if [here](https://github.com/cubicdaiya/slackboard/pull/17/files/0eb0a1d161c0d118b6eea934a3c2ad6d315e0d0b#diff-baaea793b4947ee4c469cc91abbe5c38R133) should not.